rotkehlchen/chain/ethereum/modules/curve/decoder.py

```diff
@@ -45,10 +48,33 @@
 GAUGE_DEPOSIT = b'\xe1\xff\xfc\xc4\x92=\x04\xb5Y\xf4\xd2\x9a\x8b\xfcl\xda\x04\xeb[\r<F\x07Q\xc2@,\\\\\xc9\x10\x9c'  # noqa: E501
 GAUGE_WITHDRAW = b'\x88N\xda\xd9\xceo\xa2D\r\x8aT\xcc\x124\x90\xeb\x96\xd2v\x84y\xd4\x9f\xf9\xc76a%\xa9BCd'  # noqa: E501
 
+TOKEN_EXCHANGE = b'\x8b>\x96\xf2\xb8\x89\xfaw\x1cS\xc9\x81\xb4\r\xaf\x00_c\xf67\xf1\x86\x9fppR\xd1Z=\xd9q@'  # noqa: E501
+TOKEN_EXCHANGE_UNDERLYING = b'\xd0\x13\xca#\xe7ze\x00<,e\x9cTB\xc0\x0c\x80Sq\xb7\xfc\x1e\xbdL lA\xd1Sk\xd9\x0b'  # noqa: E501
+EXCHANGE_MULTIPLE = b'\x14\xb5a\x17\x8a\xe0\xf3h\xf4\x0f\xaf\xd0H\\Oq)\xeaq\xcd\xc0\x0bL\xe1\xe5\x94\x0f\x9b\xc6Y\xc8\xb2'  # noqa: E501
+CURVE_SWAP_ROUTER = string_to_evm_address('0x99a58482BD75cbab83b27EC03CA68fF489b5788f')
+
 logger = logging.getLogger(__name__)
 log = RotkehlchenLogsAdapter(logger)
 
 
+def _read_curve_asset(
+        asset_address: Optional[ChecksumEvmAddress],
+        chain_id: ChainID,
+) -> Optional[Asset]:
```

Comment:
Technically it can, but there is no need for this, since the asset that is returned from here is used only for comparison with `event.asset` when iterating over `decoded_events`.

Reply:
Yes but isn't it better to be more specific? Since you already use the `Asset` constructor and the `evm_address_to_identifier` functions this should not be any more work and would be more explicit as to what type of asset this function returns.

I would only consider keeping it `Asset` if there is any more (considerable) work that is done with other constructors.

Okay .. just checked the code. It's not much work but it's potentially more DB calls. It's fine then. **leave it as is**
but also add a comment explaining why it's asset. That you don't use any other fields.